### PR TITLE
[blocker] Make CEB compatible with entity-8.x-0.x

### PR DESCRIPTION
**Pull request by dawehner**
- Fixes test coverage
- Fixed EntityBase to use the new Trait properly

### CODE REVIEW

**Comment by Jaesin** [comment]
Will `$fields['revision_translation_affected']` be added to `entityRevisionLogBaseFieldDefinitions()`?

Reference: https://www.drupal.org/node/2513894

**Comment by dawehner** [comment]
No, not yet, do we need this?

In the meantime this PR now contains a couple of changes to fix the tests. I'm not 100% sure about it, but adding the list operation maybe makes sense

**Comment by dawehner** [comment]
Ha, finally its even green, note, the latest commit will also be part of entity module, most probably: https://github.com/fago/entity/pull/41

**Comment by jacine** [comment]
Wondering if this still needed?

**Comment by dawehner** [comment]
@Jaesin 
Btw. you are just so right all the time, here are issues on entity module and core:

https://github.com/fago/entity/issues/45
https://www.drupal.org/node/2714163

File: src/Entity/EntityBase.php [hunk 1]
```diff
@@ -147,10 +147,7 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setDescription(t('The entity type.'))
       ->setSetting('target_type', $entity_type->getBundleEntityType());
 
-    $fields['revision_log'] = BaseFieldDefinition::create('string_long')
-      ->setLabel(t('Revision log message'))
-      ->setDescription(t('The log entry explaining the changes in this revision.'))
-      ->setRevisionable(TRUE);
+    $fields += static::entityRevisionLogBaseFieldDefinitions();
```
Comment (Jaesin) [inline src/Entity/EntityBase.php:123]: entityRevisionLogBaseFieldDefinitions adds base fields that are look to be redundant. The entity_revision table already has revision_timestamp, revision_uid and revision_log. It also looks like these fields have been removed in the 8.x-2.x verison of the entity module. This pull request also results in a duplicate log message field. Is there a reason we need these fields added to the entity_data table?

Comment (dawehner) [inline src/Entity/EntityBase.php:123]: Ah I see why you are confused. You probably used the drupal core 8.1.x ... which adds those fields automatically now, see the other PR. This PR is just about fixing some failures + getting it working with the last 8.0.x/8.x-0.x version

**Comment by Jaesin** [comment]
I think that it would be better to make CEB independent of the entity module See: https://www.drupal.org/node/2718299 . It will just cause confusion and frustration to have a complexed dependency pattern that site builders have to deal with and the entity module has already deprecated some of the classes we are using because similar classes are in core in 8.1.x. 

Instead, I've made the 1.x version of CEB more api compatible with what's in core in 8.1.x in the [small-fix](https://github.com/Jaesin/content_entity_base/tree/small-fix) branch. 

I think if more entity related features get into core in 8.2.x we can create a 8.x-2.x version of CEB and drop redundant code that was brought into core.

**Comment by heddn** [comment]
Yes. I agree. Managing the moving target of ceb, entity, core and my own
custom entity's needs had become very complex. Too many moving parts, one
might argue. So, dropping that dependency is a good thing.
On May 19, 2016 7:01 PM, "Jaesin Mulenex" notifications@github.com wrote:

I think that it would be better to make CEB independent of the entity
module See: https://www.drupal.org/node/2718299 . It will just cause
confusion and frustration to have a complexed dependency pattern that site
builders have to deal with and the entity module has already deprecated
some of the classes we are using because similar classes are in core in
8.1.x.

Instead, I've made the 1.x version of CEB more api compatible with what's
in core in 8.1.x in the small-fix
https://github.com/Jaesin/content_entity_base/tree/small-fix branch.

I think if more entity related features get into core in 8.2.x we can
create a 8.x-2.x version of CEB and drop redundant code that was brought
into core.

**Comment by Jaesin** [comment]
And finally Travis goes green.

**Pull request merged**